### PR TITLE
Land - Font optimizations - Adobe Fonts / Typekit support 

### PR DESCRIPTION
**Pull request by ijjk**
This updates this initial PR here https://github.com/vercel/next.js/pull/18146 to resolve merge conflicts and updates tests since we aren't able to update that PR itself.

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [x] Integration tests added
- [x] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.

Closes: https://github.com/vercel/next.js/pull/18146

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 15.8s | 15.6s | -218ms |
| buildDurationCached | 5.1s | 4.8s | -351ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.715 | 2.737 | ⚠️ +0.02 |
| / avg req/sec | 920.91 | 913.43 | ⚠️ -7.48 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.51 | 1.494 | -0.02 |
| /error-in-render avg req/sec | 1656.08 | 1673.74 | +17.66 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.3 kB | 19.3 kB | ✓ |
| webpack-HASH.js gzip | 996 B | 996 B | ✓ |
| Overall change | 59.6 kB | 59.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.04 kB | 3.05 kB | ⚠️ +16 B |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| 99a142a5cfae..804.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.51 kB | 8.52 kB | ⚠️ +16 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 393 B | 394 B | ⚠️ +1 B |
| Overall change | 393 B | 394 B | ⚠️ +1 B |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| index.html gzip | 561 B | 560 B | -1 B |
| link.html gzip | 570 B | 569 B | -1 B |
| withRouter.html gzip | 557 B | 557 B | ✓ |
| Overall change | 1.69 kB | 1.69 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -2,7 +2,7 @@ self.__BUILD_MANIFEST = {
   __rewrites: { beforeFiles: [], afterFiles: [], fallback: [] },
   "/": ["static\u002Fchunks\u002Fpages\u002Findex-781a77dd168190130aad.js"],
   "/_error": [
-    "static\u002Fchunks\u002Fpages\u002F_error-9459a67ba367dde23f4d.js"
+    "static\u002Fchunks\u002Fpages\u002F_error-77a136c514089a3e89cf.js"
   ],
   "/amp": ["static\u002Fchunks\u002Fpages\u002Famp-a0345799be7f0667d737.js"],
   "/css": [
```
</details>
<details>
<summary>Diff for <strong>_error-HASH.js</strong></summary>

```diff
@@ -440,7 +440,10 @@ Also adds support for deduplicated `key` properties
               if (
                 c.type === "link" &&
                 c.props["href"] && // TODO(prateekbh@): Replace this with const from `constants` when the tree shaking works.
-                ["https://fonts.googleapis.com/css"].some(function(url) {
+                [
+                  "https://fonts.googleapis.com/css",
+                  "https://use.typekit.net/"
+                ].some(function(url) {
                   return c.props["href"].startsWith(url);
                 })
               ) {
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -48,7 +48,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -48,7 +48,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 19.4s | 19.1s | -335ms |
| buildDurationCached | 7.1s | 6.8s | -253ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.3 kB | 19.3 kB | ✓ |
| webpack-HASH.js gzip | 996 B | 996 B | ✓ |
| Overall change | 59.6 kB | 59.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.04 kB | 3.05 kB | ⚠️ +16 B |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| 99a142a5cfae..804.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.51 kB | 8.52 kB | ⚠️ +16 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 393 B | 394 B | ⚠️ +1 B |
| Overall change | 393 B | 394 B | ⚠️ +1 B |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _error.js | 1.34 MB | 1.34 MB | ⚠️ +56 B |
| 404.html | 2.42 kB | 2.42 kB | ✓ |
| 500.html | 2.41 kB | 2.41 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.61 kB | 1.61 kB | ✓ |
| css.html | 1.79 kB | 1.79 kB | ✓ |
| hooks.html | 1.67 kB | 1.67 kB | ✓ |
| index.js | 1.35 MB | 1.35 MB | ⚠️ +56 B |
| link.js | 1.4 MB | 1.4 MB | ⚠️ +56 B |
| routerDirect.js | 1.39 MB | 1.39 MB | ⚠️ +56 B |
| withRouter.js | 1.39 MB | 1.39 MB | ⚠️ +56 B |
| Overall change | 6.9 MB | 6.9 MB | ⚠️ +280 B |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 12.8s | 13.1s | ⚠️ +260ms |
| buildDurationCached | 5.5s | 5.4s | -90ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.712 | 2.659 | -0.05 |
| / avg req/sec | 921.83 | 940.04 | +18.21 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.512 | 1.508 | 0 |
| /error-in-render avg req/sec | 1652.89 | 1657.67 | +4.78 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.19 kB | 7.19 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-HASH.js gzip | 3.71 kB | 3.72 kB | ⚠️ +14 B |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 339 B | 339 B | ✓ |
| hooks-HASH.js gzip | 887 B | 887 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 303 B | 303 B | ✓ |
| withRouter-HASH.js gzip | 302 B | 302 B | ✓ |
| 21c68fa65a48..217.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.36 kB | 9.37 kB | ⚠️ +14 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -2,7 +2,7 @@ self.__BUILD_MANIFEST = {
   __rewrites: { beforeFiles: [], afterFiles: [], fallback: [] },
   "/": ["static\u002Fchunks\u002Fpages\u002Findex-b460df3d63326fbb06a1.js"],
   "/_error": [
-    "static\u002Fchunks\u002Fpages\u002F_error-c75557f170c57b2c8979.js"
+    "static\u002Fchunks\u002Fpages\u002F_error-176a8b2ad3a2229a0373.js"
   ],
   "/amp": ["static\u002Fchunks\u002Fpages\u002Famp-27f75ad11120c5cdedd1.js"],
   "/css": [
```
</details>
<details>
<summary>Diff for <strong>_error-HASH.js</strong></summary>

```diff
@@ -544,7 +544,10 @@ Also adds support for deduplicated `key` properties
               if (
                 c.type === "link" &&
                 c.props["href"] && // TODO(prateekbh@): Replace this with const from `constants` when the tree shaking works.
-                ["https://fonts.googleapis.com/css"].some(function(url) {
+                [
+                  "https://fonts.googleapis.com/css",
+                  "https://use.typekit.net/"
+                ].some(function(url) {
                   return c.props["href"].startsWith(url);
                 })
               ) {
```
</details>
</details>
Commit: fe9ce86a5bfe5cf39d50e77357c90b761228765b

**Review by shuding — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 12.2s | 12.3s | ⚠️ +79ms |
| buildDurationCached | 4.1s | 3.8s | -319ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.051 | 2.041 | -0.01 |
| / avg req/sec | 1218.79 | 1224.87 | +6.08 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.131 | 1.082 | -0.05 |
| /error-in-render avg req/sec | 2209.72 | 2309.61 | +99.89 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.3 kB | 19.3 kB | ✓ |
| webpack-HASH.js gzip | 996 B | 996 B | ✓ |
| Overall change | 59.6 kB | 59.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.04 kB | 3.05 kB | ⚠️ +16 B |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| 99a142a5cfae..804.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.51 kB | 8.52 kB | ⚠️ +16 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 393 B | 394 B | ⚠️ +1 B |
| Overall change | 393 B | 394 B | ⚠️ +1 B |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| index.html gzip | 561 B | 560 B | -1 B |
| link.html gzip | 570 B | 569 B | -1 B |
| withRouter.html gzip | 557 B | 557 B | ✓ |
| Overall change | 1.69 kB | 1.69 kB | -2 B |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -2,7 +2,7 @@ self.__BUILD_MANIFEST = {
   __rewrites: { beforeFiles: [], afterFiles: [], fallback: [] },
   "/": ["static\u002Fchunks\u002Fpages\u002Findex-781a77dd168190130aad.js"],
   "/_error": [
-    "static\u002Fchunks\u002Fpages\u002F_error-9459a67ba367dde23f4d.js"
+    "static\u002Fchunks\u002Fpages\u002F_error-77a136c514089a3e89cf.js"
   ],
   "/amp": ["static\u002Fchunks\u002Fpages\u002Famp-a0345799be7f0667d737.js"],
   "/css": [
```
</details>
<details>
<summary>Diff for <strong>_error-HASH.js</strong></summary>

```diff
@@ -440,7 +440,10 @@ Also adds support for deduplicated `key` properties
               if (
                 c.type === "link" &&
                 c.props["href"] && // TODO(prateekbh@): Replace this with const from `constants` when the tree shaking works.
-                ["https://fonts.googleapis.com/css"].some(function(url) {
+                [
+                  "https://fonts.googleapis.com/css",
+                  "https://use.typekit.net/"
+                ].some(function(url) {
                   return c.props["href"].startsWith(url);
                 })
               ) {
```
</details>
<details>
<summary>Diff for <strong>index.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -48,7 +48,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>link.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -53,7 +53,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
<details>
<summary>Diff for <strong>withRouter.html</strong></summary>

```diff
@@ -7,7 +7,7 @@
     <noscript data-n-css=""></noscript>
     <link
       rel="preload"
-      href="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      href="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       as="script"
     />
     <link
@@ -48,7 +48,7 @@
       src="/_next/static/chunks/polyfills-01e9caf46d9a7f038c09.js"
     ></script>
     <script
-      src="/_next/static/chunks/webpack-dd6fffdf0dbc836f25a7.js"
+      src="/_next/static/chunks/webpack-d70f99a1c072ba84a76b.js"
       async=""
     ></script>
     <script
```
</details>
</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 15s | 14.9s | -87ms |
| buildDurationCached | 5.7s | 5.8s | ⚠️ +57ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| framework-HASH.js gzip | 39.3 kB | 39.3 kB | ✓ |
| main-HASH.js gzip | 19.3 kB | 19.3 kB | ✓ |
| webpack-HASH.js gzip | 996 B | 996 B | ✓ |
| Overall change | 59.6 kB | 59.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.02 kB | 1.02 kB | ✓ |
| _error-HASH.js gzip | 3.04 kB | 3.05 kB | ⚠️ +16 B |
| amp-HASH.js gzip | 526 B | 526 B | ✓ |
| css-HASH.js gzip | 334 B | 334 B | ✓ |
| hooks-HASH.js gzip | 890 B | 890 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| routerDirect..HASH.js gzip | 331 B | 331 B | ✓ |
| withRouter-HASH.js gzip | 329 B | 329 B | ✓ |
| 99a142a5cfae..804.css gzip | 125 B | 125 B | ✓ |
| Overall change | 8.51 kB | 8.52 kB | ⚠️ +16 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 393 B | 394 B | ⚠️ +1 B |
| Overall change | 393 B | 394 B | ⚠️ +1 B |

</details>

<details>
<summary><strong>Serverless bundles</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _error.js | 1.34 MB | 1.34 MB | ⚠️ +56 B |
| 404.html | 2.42 kB | 2.42 kB | ✓ |
| 500.html | 2.41 kB | 2.41 kB | ✓ |
| amp.amp.html | 10.8 kB | 10.8 kB | ✓ |
| amp.html | 1.61 kB | 1.61 kB | ✓ |
| css.html | 1.79 kB | 1.79 kB | ✓ |
| hooks.html | 1.67 kB | 1.67 kB | ✓ |
| index.js | 1.35 MB | 1.35 MB | ⚠️ +56 B |
| link.js | 1.4 MB | 1.4 MB | ⚠️ +56 B |
| routerDirect.js | 1.39 MB | 1.39 MB | ⚠️ +56 B |
| withRouter.js | 1.39 MB | 1.39 MB | ⚠️ +56 B |
| Overall change | 6.9 MB | 6.9 MB | ⚠️ +280 B |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| buildDuration | 10.6s | 10.6s | -14ms |
| buildDurationCached | 4.3s | 4.3s | -41ms |
| nodeModulesSize | 46.6 MB | 46.6 MB | ⚠️ +158 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.098 | 2.103 | ⚠️ +0.01 |
| / avg req/sec | 1191.72 | 1188.63 | ⚠️ -3.09 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.118 | 1.108 | -0.01 |
| /error-in-render avg req/sec | 2236.49 | 2255.35 | +18.86 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| 677f882d2ed8..HASH.js gzip | 13.3 kB | 13.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-HASH.js gzip | 7.19 kB | 7.19 kB | ✓ |
| webpack-HASH.js gzip | 751 B | 751 B | ✓ |
| Overall change | 60.2 kB | 60.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-HASH.js gzip | 3.71 kB | 3.72 kB | ⚠️ +14 B |
| amp-HASH.js gzip | 536 B | 536 B | ✓ |
| css-HASH.js gzip | 339 B | 339 B | ✓ |
| hooks-HASH.js gzip | 887 B | 887 B | ✓ |
| index-HASH.js gzip | 227 B | 227 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 303 B | 303 B | ✓ |
| withRouter-HASH.js gzip | 302 B | 302 B | ✓ |
| 21c68fa65a48..217.css gzip | 125 B | 125 B | ✓ |
| Overall change | 9.36 kB | 9.37 kB | ⚠️ +14 B |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| _buildManifest.js gzip | 420 B | 420 B | ✓ |
| Overall change | 420 B | 420 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js feat/typekit-support | Change |
| - | - | - | - |
| index.html gzip | 613 B | 613 B | ✓ |
| link.html gzip | 620 B | 620 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

#### Diffs
<details>
<summary>Diff for <strong>_buildManifest.js</strong></summary>

```diff
@@ -2,7 +2,7 @@ self.__BUILD_MANIFEST = {
   __rewrites: { beforeFiles: [], afterFiles: [], fallback: [] },
   "/": ["static\u002Fchunks\u002Fpages\u002Findex-b460df3d63326fbb06a1.js"],
   "/_error": [
-    "static\u002Fchunks\u002Fpages\u002F_error-c75557f170c57b2c8979.js"
+    "static\u002Fchunks\u002Fpages\u002F_error-176a8b2ad3a2229a0373.js"
   ],
   "/amp": ["static\u002Fchunks\u002Fpages\u002Famp-27f75ad11120c5cdedd1.js"],
   "/css": [
```
</details>
<details>
<summary>Diff for <strong>_error-HASH.js</strong></summary>

```diff
@@ -544,7 +544,10 @@ Also adds support for deduplicated `key` properties
               if (
                 c.type === "link" &&
                 c.props["href"] && // TODO(prateekbh@): Replace this with const from `constants` when the tree shaking works.
-                ["https://fonts.googleapis.com/css"].some(function(url) {
+                [
+                  "https://fonts.googleapis.com/css",
+                  "https://use.typekit.net/"
+                ].some(function(url) {
                   return c.props["href"].startsWith(url);
                 })
               ) {
```
</details>
</details>
Commit: c074d0ceeee8341da73552f089bebd76159ed0da

File: test/integration/font-optimization/fixtures/with-typekit/server.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
+const http = require('http')
```
Comment (janicklas-ralph) [inline test/integration/font-optimization/fixtures/with-typekit/server.js:1]: We don't need this file to be duplicated

**Comment by andrewgadziksonos** [comment]
This is exciting! Thanks for the help!

**Pull request merged**